### ISSUE TITLE
Make license renewal failure non-fatal until the end of the license duration

### ISSUE THREAD

**Issue by noamtamim** [google/ExoPlayer#6291]
### [REQUIRED] Use case description
When the license is **really** expired we don't even pass the stream to ExoPlayer, we fire an error event instead. Now we also have to take this value into account, but it's a private constant (in DefaultDrmSession) so we can't. Hardcoding `60` (the current constant value) in our code seems like a bad idea because this value may change, at some point.

### Proposed solution
Move this value to the almighty C class (as public) - then we can read it.

### Alternatives considered
Hardcode this value in application code.

**Comment by AquilesCanta** [comment]
> When the license is really expired we don't even pass the stream to ExoPlayer

Could you clarify what this means? Also could you provide a precise example of the required usecase?

**Comment by noamtamim** [comment]
Let's say a downloaded asset has expired. 


When a downloaded asset expires, the UI control to play it gets disabled or replaced with a "renew" button. In any case, the app avoids "expired" errors by simply not trying to play expired assets.

With MAX_LICENSE_DURATION_TO_RENEW_SECONDS, effectively the expiration is 60-seconds shorter. If there are 59 seconds remaining and the user taps "play", the app doesn't consider the asset to be expired and it tries to play. But then ExoPlayer prevents playback, and an error has to be shown to the user.

If you make it public, we can take this value into account and consider the asset expired.

**Comment by AquilesCanta** [comment]
> But then ExoPlayer prevents playback, and an error has to be shown to the user.

Do you mean that the license renewal fails?

**Comment by noamtamim** [comment]
When you play offline assets you may actually be offline, so renewal is not possible. And there's still one minute left, so renewal is not required, either.

**Comment by AquilesCanta** [comment]
If I understand correctly, your issue is that you want to prevent offline content from playing whenever `license remaining duration - 60 < duration of the content` (or a similar condition). The reason being that the DefaultDrmSession tries to renew the offline license 60 seconds from the end of the license duration and, when it fails, an error is propagated. Please confirm whether this is true.

If it is indeed the case, I am tempted to rather make the license renewal failure non-fatal until it's actually fatal, which is at the end of the license duration. Please confirm so I can discuss with the rest of the team about which approach makes more sense.

**Comment by noamtamim** [comment]
Yes -- I believe your proposal makes sense. If renewal failure is not fatal, playback can start.

**Comment by ojw28** [comment]
Could we update the title of this issue to track the proposal? Thanks!

**Comment by noamtamim** [comment]
I believe the title should be something like "make the license renewal failure non-fatal until the end of the license duration". But isn't the proposal different enough from the original request as to require a new ticket?

**Comment by ojw28** [comment]
I don't mind either way, as long as there's one issue that clearly states what we're planning to do :). Please close this one and file a new one, if it's easier. Thanks!

**Comment by AquilesCanta** [comment]
Updated the title. A commit will appear below once we submit one. In the meantime you can hard-code the constant, until we get a few free cycles to get this done.
